README.md

```diff
@@ -156,393 +156,982 @@ The executor MUST be the UCAN delegate. Their DID MUST be set the in `aud` field
+          "title": "How UCAN Tasks Changed My Life",
+          "body": "This is the story of how one spec changed everything...",
+          "topics": ["authz", "journal"],
+          "draft": true
+        }
+      }
+    }
+  ],
+  "prf": [{"/": "bafkreibbz5pksvfjyima4x4mduqpmvql2l4gh5afaj4ktmw6rwompxynx4"}],
+  "sig": {"/": {"bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"}}
+}
+```
+
+# 8 Result
+
+A Result records the output of a Task, as well as its success or failure state.
+
+## 8.1 Variants
 
 ``` ipldsch
-type TaskPointer struct {
-  inv       InvocationPointer
-  taskLabel String
-} representation tuple
+type Result union {
+  | Any ("ok")  -- Success
+  | Any ("err") -- Failure
+} representation keyed
+```
 
-type InvocationPointer union {
-  | Local "/"
-  | &SignedInvocation
+### 8.1.1 Success
+
+The `val` field MUST contain the value returned from a successful Task. The exact shape of the returned data is left undefined to allow for flexibility in various Task types.
+
+``` json
+{"ok": 42}
+```
+
+### 8.1.2 Failure
+
+The `err` branch MAY contain detail about why execution failed. It is left undefined in this specification to allow for Task types to standardize the data that makes sense in their conetxts.
+
+If no information is available, this field SHOULD be set to `Null`.
+
+``` json
+{
+  "err": {
+    "dev/reason": "unauthorized",
+    "http/status": 401
+  }
 }
 ```
 
-## 5.3 JSON Examples
+# 9 Receipt
+
+An Invocation Receipt is an attestation of the Result of an Invocation. A Receipt MUST be signed by the Executor (the `aud` of the associated UCANs).
+
+**NB: a Receipt this does not guarantee correctness of the result!** The statement's veracity MUST be only understood as an attestation from the executor.
+
+Receipts MUST use the same version as the invocation that they contain.
+
+## 9.1 Fields
+
+``` ipldsch
+type Receipt struct {
+  ran  Pointer
+  out  {String : Result}
+  rec  {String : &Receipt} -- FIXME describe below
+  meta {String : Any}
+  sig  Varsig
+} 
+```
+// FIXME #### 8.1.2.2 Proof
+//
+/ / In the case that the Invocation was delegated to another Executor, a `prf` containing a [`Receipt`](FIXME) MAY be supplied as proof of the `Result`.
+//
+
+### 9.1.1 Task
+
+The `inv` field MUST include a link to the Task that the Receipt is for.
+
+### 9.1.2 Output
+
+The `out` field MUST contain the output of steps of the call graph, indexed by the task name inside the invocation. If the invocation is the implicit `"*"`, then the base64 hash of the concatenation of the URI, Ability and extensional fields MUST be used.
+
+The `out` field MAY omit any tasks that have not yet completed, or results which are not public. An `Task` may be associated to zero or more `Receipts`.
+
+A `Result` MAY include recursive `Receipt` CIDs in on the `Success` branch. As a Task may require subdelegation, the OPTIONAL `rec` field MAY be used to include recursive `Receipt`s.
```

Comment:
Perfect! Was going to make a comment to this effect on the previous version but you are way ahead as ever.

Why the base64 hash? Why not just the `Task` CID, or even better, the "memoization CID" ignoring the meta fields?

Reply:
@simonwo, ah indeed I need to update this to use the Closure CID now that it's a thing. Will fix 👍